src/views/CardStack.js

```diff
@@ -189,18 +182,27 @@ class CardStack extends Component<DefaultProps, Props, void> {
     transitionProps: NavigationTransitionProps,
     headerMode: HeaderMode
   ): ?React.Element<*> {
-    const headerConfig = this.props.router.getScreenConfig(
+    const headerConfig: HeaderConfig = this.props.router.getScreenConfig(
       transitionProps.navigation,
       'header'
     ) || {};
 
+    if (
+      headerConfig.bar === null
+      || (headerConfig.bar && typeof headerConfig.bar !== 'function')
+    ) {
+      return headerConfig.bar;
+    }
+
+    const HeaderComponent = headerConfig.bar || Header;
+
     return (
-      <this.props.headerComponent
+      <HeaderComponent
         {...transitionProps}
         router={this.props.router}
         style={headerConfig.style}
         mode={headerMode}
-        onNavigateBack={() => this.props.navigation.goBack(null)}
+        onNavigateBack={() => { this.props.navigation.goBack(null); }}
```

Comment:
Required, otherwise flow complains that `goBack` returns `boolean` instead of `void` expected. Turns out it started working after I annotated `HeaderConfig` few lines above.